src/configure/configure.ts

```diff
@@ -276,7 +338,19 @@ class PipelineConfigurer {
 
     private async getAzureDevOpsDetails(): Promise<void> {
         try {
-            if (this.inputs.sourceRepository.repositoryProvider !== RepositoryProvider.AzureRepos) {
+            if (this.inputs.sourceRepository.repositoryProvider === RepositoryProvider.AzureRepos) {
+                let orgAndProjectName = AzureDevOpsHelper.getOrganizationAndProjectNameFromRepositoryUrl(this.inputs.sourceRepository.remoteUrl);
+                this.inputs.organizationName = orgAndProjectName.orgnizationName;
+                this.azureDevOpsClient.getRepository(this.inputs.organizationName, orgAndProjectName.projectName, this.inputs.sourceRepository.repositoryName)
+                    .then((repository) => {
+                        this.inputs.sourceRepository.repositoryId = repository.id;
```

Comment:
Does not look right to use this.inputs as storage. We can create a separate Context definition which acts as local storage and then pass it along.

Reply:
You can consider this comment later as well, no need to do now

Reply:
yes sounds like it will align once we do a design review.